analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultSensorContext.java

```diff
@@ -188,4 +190,24 @@ private static UnsupportedOperationException unsupported() {
     return new UnsupportedOperationException("Not supported in SonarLint");
   }
 
+  @Override
+  public boolean canSkipUnchangedFiles() {
+    return false;
+  }
+
+  @Override
+  public boolean isCacheEnabled() {
+    return false;
+  }
+
+  @Override
+  public ReadCache previousCache() {
+    throw unsupported();
```

Comment:
Will this be conditionally called on the analyzers side ? Else we could have a no-op read and write cache

Reply:
Yeah, normally they should call it only when `isCacheEnabled` returns `true`